app/forms/what_county_do_you_live_in_form.rb

```diff
@@ -0,0 +1,24 @@
+class WhatCountyDoYouLiveInForm < Form
+  set_attributes_for :navigator, :county
+
+  validates_presence_of :county
+
+  def save
+    unless report.present?
+      self.report = Report.create
```

Comment:
There are now two different places where the Report is created, here and the zip code page. This led to all kinds of problems on GetCalFresh. Any suggestions for how to combine the controller? Perhaps using partials and url param?

Reply:
We could make a shared method on `Form` called `create_report`. It would essentially be this `unless ...` block.

Then from any child form that needs it like this one, we could call:

`report = create_report`

Reply:
It's less about drying up the code, more about having multiple paths we need to support. I remember we had the same thing in GCF, then partnerships wanted to run an a/b/c test that was editing one of the county location pages. We had to support four different places where the  app really starts. It got too complicated to easily change, especially when tracking funnels, etc.